nuxeo-drive-client/nxdrive/wui/translator.py

```diff
@@ -1,85 +1,78 @@
-'''
-@author: Remi Cattiau
-'''
+# coding: utf-8
+import codecs
 import json
 import re
 
 
 class Translator(object):
-    '''
-    classdocs
-    '''
 
     _singleton = None
 
     def __init__(self, manager, filepath, lang=None):
-        '''
-        Constructor
-        '''
         self._labels = None
         self._manager = manager
+
         # Load from JSON
-        with open(filepath, "r") as fp:
-            json_raw = fp.read()
-        # Remove the setter for use in Angular
-        json_raw = json_raw.replace("LABELS=", "")
-        self._labels = json.loads(json_raw, 'utf-8')
+        with codecs.open(filepath, encoding='utf-8') as fp:
+            self._labels = json.loads(fp.read().lstrip('LABELS='))
+
         # List language
         self._langs = dict()
         for key in self._labels:
-            if not "LANGUAGE" in self._labels[key]:
-                continue
-            self._langs[key] = (key, self._labels[key]["LANGUAGE"])
+            try:
+                self._langs[key] = (key, self._labels[key]['LANGUAGE'])
+            except KeyError:
+                pass
+
         # Select one
-        if lang is None or not lang in self._langs:
-            if "en" in self._labels:
-                self._set("en")
-                self._fallback = self._labels["en"]
-            else:
-                self._set(iter(self._labels).next())
-                self._fallback = dict()
-        else:
+        try:
             self._set(lang)
-            if "en" in self._langs:
-                self._fallback = self._labels["en"]
+        except ValueError:
+            self._set('en')
+        self._fallback = self._labels['en']
         Translator._singleton = self
 
-    def _tokenize(self, label, values=None):
-        if values is None:
+    @staticmethod
+    def _tokenize(label, values=None):
+        if not values:
             return label
+
         result = label
-        for token in re.findall(r'\{\{[^\}]+\}\}', label):
+        for token in re.findall(r'{{[^}]+}}', label):
             attr = token[2:-2].strip()
-            value = ""
-            if attr in values:
+            try:
```

Comment:
I prefer the idiomatic EAFP (Easier to Ask for Forgiveness than Permission) in general. Here, it will prevent twice lookups for instance.